src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -34,6 +34,8 @@ internal static class IntrinsicFunctions
 
         private static readonly object[] DefaultRegistryViews = new object[] { RegistryView.Default };
 
+        private static readonly Lazy<Regex> RegistrySdkRegex = new Lazy<Regex>(() => new Regex(@"^HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Microsoft SDKs\\Windows\\v(\d+\.\d+)$", RegexOptions.Compiled | RegexOptions.IgnoreCase));
```

Comment:
Nice! Can you included in the commit notes a ballpark of how often this ran? And did you test against a static but not compiled version? Compilation takes enough time that the better runtime perf isn't always a clean win :(

Reply:
It was called hundreds of time, so I just assumed compilation would be better at that point. I'll undo that commit and run the perf tests again, to see how it moves the needle.

Reply:
"hundreds of times" is enough for me :)

Reply:
Turns out the regex is never called at all on old and new style csprojes! So I left the static Regex, but removed the compiled flag just in case.